lib/cfme/cloud_services/data_collector.rb

```diff
@@ -92,11 +96,48 @@ def extract_data(target, model_manifest)
     end
   end
 
+  def send_via_http_request(payload)
+    begin
+      $log.info "Trying to send inventory thru HTTP request..."
+      request = Net::HTTP::Post.new(UPLOADER_URI, header)
+      request.body = request_body(payload)
+
+      http = Net::HTTP.new(UPLOADER_HOST, UPLOADER_PORT)
+      result = http.request(request)
+      $log.info "Request send sucessfully (#{result.class}: #{result.value})"
+      true
+    rescue StandardError => e
+      $log.error "Connection error: #{e}"
+      false
+    end
+  end
+
+  INSIGHTS_CLIENT_COMMAND = "insights-client".freeze
+
+  def send_via_insights_client(payload)
+    $log.info "Trying to send inventory thru #{INSIGHTS_CLIENT_COMMAND}..."
+    status = nil
+
+    packed_file_from(gzipped_tar_content_from(payload)) do |packed_temporary_file_path|
+      command_params = %W[--payload=#{packed_temporary_file_path} --content-type=#{CONTENT_TYPE}]
+      stdout, stderr, status = Open3.capture3(INSIGHTS_CLIENT_COMMAND, *command_params)
```

Comment:
You can use AwesomeSpawn which wraps a bunch of this logic rather than calling `Open3` directly

Reply:
done 👍